packages/pigeon/platform_tests/alternate_language_test_plugin/android/src/test/java/com/example/alternate_language_test_plugin/AllDatatypesTest.java

```diff
@@ -142,10 +141,11 @@ public void hasValues() {
   public void integerToLong() {
     AllTypes everything = new AllTypes();
     everything.setAnInt(123L);
-    Map<String, Object> map = everything.toMap();
-    assertTrue(map.containsKey("anInt"));
-    map.put("anInt", 123);
-    AllTypes readEverything = AllTypes.fromMap(map);
+    ArrayList<Object> list = everything.toList();
+    assertNotNull(list);
+    assertNull(list.get(0));
+    assertNotNull(list.get(1));
+    AllTypes readEverything = AllTypes.fromList(list);
```

Comment:
This isn't testing the same thing.  I can't remember off the top of my head but this is asserting some magic related to how numbers are handled in the encoder.

Reply:
This is reminding me that I was planning on doing my own review to find lost in translation issues like this one.